gap/PolygonalComplexes/properties.gi

```diff
@@ -218,6 +218,141 @@ InstallMethod( FaceCounter, "for a twisted polygonal complex",
         return Collected( Compacted( faceDegrees ) );
     end
 );
+
+InstallMethod( ButterflyCounter, "for a simplicial surface",
+    [IsSimplicialSurface],
+    function(surface)
+	local e,temp2,counter,g,VerticesOfOrthogonalEdge,voe1,voe2,degOfVert1,degOfVert2;
+
+        VerticesOfOrthogonalEdge:=function(S,e)
+            local temp;
+            temp:=FacesOfEdge(S,e);
+            temp:=Union(VerticesOfFace(S,temp[1]),VerticesOfFace(S,temp[2]));
+            return Difference(temp, VerticesOfEdge(S,e));
+	end;
+	counter:=[];
+	for e in InnerEdges(surface) do 
+		voe1:=VerticesOfEdge(surface,e);
+		degOfVert1:=SortedList(List(voe1,g->FaceDegreeOfVertex(surface,g)));
+		voe2:=VerticesOfOrthogonalEdge(surface,e);
+		degOfVert2:=SortedList(List(voe2,g->FaceDegreeOfVertex(surface,g)));
+		Add(counter,[degOfVert1,degOfVert2]);
+	od;
+	return Set(Collected(counter));
+end
+);
+
+
+InstallMethod( UmbrellaCounter, "for a simplicial surface",
+    [IsSimplicialSurface],
+    function(surface)
+	local n,counter,v,temp,orb,tup,G,perm,OrbitOnList,tempcounter,temp1,rightumb,bub;
+	tempcounter:=[];
+	if not IsClosedSurface(surface) then 
+		return fail;
+	fi;
+	rightumb:=function(L)
```

Comment:
You've never used L in the method, but how you use the method, L and temp are equivalent. Maybe you replace temp by L.